File: backend/app/schemas/checkout.py
from pydantic import BaseModel
from typing import List
from app.schemas.order import OrderItemOut
from app.core.enums import PaymentMethod

class CheckoutSummary(BaseModel):
    items: List[OrderItemOut]
    subtotal: int

class CheckoutRequest(BaseModel):
    shipping_address: str
    payment_method: PaymentMethod

class CheckoutResponse(BaseModel):
    order_id: int
    status: str # e.g., "confirmed", "processing", "pending"
    message: str

File: frontend/src/types/checkout.ts
import type { CartItem } from "./cart";
import type { OrderItemOut } from "./order";

export type PaymentMethod =
    | "credit_card"
    | "debit_card"
    | "bank_transfer"
    | "cash";

export interface CheckoutSummary {
    items: OrderItemOut[];
    subtotal: number;
}

export interface CheckoutRequest {
    shipping_address: string;
    payment_method: PaymentMethod;
}

export interface CheckoutResponse {
    order_id: number;
    status: string;
    message: string;
}

export interface CheckoutItem extends CartItem {
    modelId?: number;
}

export interface CheckoutItemProps {
    item: CheckoutItem;
}

export interface CheckoutFormData {
    shippingAddress: string;
    paymentMethod: PaymentMethod;
}

export interface CheckoutFormProps {
    formData: CheckoutFormData;
    onFormChange: (data: CheckoutFormData) => void;
    loading?: boolean;
}

export interface CheckoutOrderSummaryProps {
    items: CheckoutItem[];
    subtotal: number;
    tax: number;
    total: number;
    onPlaceOrder: () => void;
    loading?: boolean;
}
